Add spl-token balance field

anshuman-dev opened the pull request.

anshuman-dev committed:
Add spl-token balance field

abdelhakeem commented on components/WalletDetails.tsx:
```diff
@@ -109,23 +110,19 @@ const WalletDetails = ({ initialWallet, members }: WalletDetailsProps) => {
+    tokenAccounts.value.forEach((e) => {
+    const accountInfo = AccountLayout.decode(e.account.data);
+    setsplbalance(accountInfo.amount)
+  })
```
Here, instead of fetching _all_ the token accounts of the Hydra wallet and calling `setsplbalance` for the balance of each, you'll only need to fetch the balance of the associated token account for the specific SPL token that we accept, you can obtain its address like here: https://github.com/NikhilSharma03/hydra-keys/blob/8ce2c05b463ed478152505dc297884e4565a06dc/components/FundWalletModal.tsx#L87-L91

Then you can use it with [`getAccount`](https://solana-labs.github.io/solana-program-library/token/js/modules.html#getAccount) from `@solana/spl-token` to fetch the account.

anshuman-dev committed:
Correct Logic

abdelhakeem committed:
Use getTokenAccountBalance for SPL token balance

- Use Connection.getTokenAccountBalance for SPL token balance to get the
decimals along with the raw balance.
- Convert raw token balance to real token balance accounting for decimals.

abdelhakeem committed:
Fetch token balance only if wallet accepts SPL

abdelhakeem committed:
Refresh token balance on update

abdelhakeem committed:
fixup! Refresh token balance on update

abdelhakeem commented:
I've made a few additions and merged this.
